libcxx/test/std/strings/basic.string/string.nonmembers/string_op+/string.string_view.pass.cpp

```diff
@@ -0,0 +1,155 @@
+    LIBCPP_ASSERT(is_string_asan_correct(st + sv));
+  }
+  // string_view + string&
+  {
+    std::basic_string_view<CharT, TraitsT> sv{x};
+    std::basic_string<CharT, TraitsT, AllocT> st{y, allocator};
+
+    std::same_as<std::basic_string<CharT, TraitsT, AllocT>> decltype(auto) result = sv + st;
+    assert(result == expected);
+    assert(result.get_allocator() == allocator);
+    LIBCPP_ASSERT(is_string_asan_correct(sv + st));
+  }
+  // string_view + const string&
+  {
+    std::basic_string_view<CharT, TraitsT> sv{x};
+    const std::basic_string<CharT, TraitsT, AllocT> st{y, allocator};
+
+    std::same_as<std::basic_string<CharT, TraitsT, AllocT>> decltype(auto) result = sv + st;
+    assert(result == expected);
+    assert(result.get_allocator() == allocator);
+    LIBCPP_ASSERT(is_string_asan_correct(sv + st));
+  }
+  // string_view + string&&
+  {
+    std::basic_string_view<CharT, TraitsT> sv{x};
+    std::basic_string<CharT, TraitsT, AllocT> st{y, allocator};
+
+    std::same_as<std::basic_string<CharT, TraitsT, AllocT>> decltype(auto) result = sv + std::move(st);
+    assert(result == expected);
+    assert(result.get_allocator() == allocator);
+    LIBCPP_ASSERT(is_string_asan_correct(sv + st));
+  }
+}
+
+template <typename CharT, typename TraitsT = std::char_traits<CharT>, typename AllocT = std::allocator<CharT>>
+constexpr void test() {
+  // Concatinate with an empty `string`/`string_view`
+  test<CharT, TraitsT, AllocT>(CS(""), CS(""), CS(""));
+  test<CharT, TraitsT, AllocT>(CS(""), CS("short"), CS("short"));
+  test<CharT, TraitsT, AllocT>(CS(""), CS("not so short"), CS("not so short"));
+  test<CharT, TraitsT, AllocT>(CS(""), CS("this is a much longer string"), CS("this is a much longer string"));
+
+  test<CharT, TraitsT, AllocT>(CS(""), CS(""), CS(""));
+  test<CharT, TraitsT, AllocT>(CS("short"), CS(""), CS("short"));
+  test<CharT, TraitsT, AllocT>(CS("not so short"), CS(""), CS("not so short"));
+  test<CharT, TraitsT, AllocT>(CS("this is a much longer string"), CS(""), CS("this is a much longer string"));
+
+  // Non empty
+  test<CharT, TraitsT, AllocT>(CS("B"), CS("D"), CS("BD"));
+  test<CharT, TraitsT, AllocT>(CS("zmt94"), CS("+hkt82"), CS("zmt94+hkt82"));
+  test<CharT, TraitsT, AllocT>(CS("not so short"), CS("+is not bad"), CS("not so short+is not bad"));
+  test<CharT, TraitsT, AllocT>(
+      CS("this is a much longer string"),
+      CS("+which is so much better"),
+      CS("this is a much longer string+which is so much better"));
+}
+
+constexpr bool test() {
+  test<char, std::char_traits<char>>();
+  test<char, std::char_traits<char>, min_allocator<char>>();
+  test<char, std::char_traits<char>, safe_allocator<char>>();
+
+#ifndef TEST_HAS_NO_WIDE_CHARACTERS
+  test<wchar_t, std::char_traits<wchar_t>>();
+  test<wchar_t, std::char_traits<wchar_t>, min_allocator<wchar_t>>();
+  test<wchar_t, std::char_traits<wchar_t>, safe_allocator<wchar_t>>();
+#endif
```

Comment:
Can you test with the following additional types `char8_t`, `char16_t`, and `char32_t`?
Maybe it would be good to have another layer of indirection with a templated function for  `class CharT`